src/main/java/br/com/vayu/services/CategoryService.java

```diff
@@ -0,0 +1,61 @@
+package br.com.vayu.services;
+
+import br.com.vayu.dto.CategoryDTO;
+import br.com.vayu.dto.CategoryFormDTO;
+import br.com.vayu.dto.CategoryManagementDTO;
+import br.com.vayu.models.Category;
+import br.com.vayu.repositories.CategoryRepository;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+@Service
+public class CategoryService {
+
+    private final CategoryRepository categoryRepository;
+
+    public CategoryService(CategoryRepository categoryRepository) {
+        this.categoryRepository = categoryRepository;
+    }
+
+    public List<CategoryDTO> getActiveCategoryDtoList() {
+        var activeCategories = categoryRepository.findAllByActiveTrue();
+
+        return activeCategories.stream()
+                .map(CategoryDTO::new)
+                .toList();
+    }
+
+    public List<CategoryManagementDTO> getCategoryManagementDtoListInOrder() {
+        return categoryRepository.findAllByOrderByOrder().stream()
+                .map(CategoryManagementDTO::new)
+                .toList();
+    }
+
+    public CategoryFormDTO getByCode(String categoryCode) {
+        var categoryModel = categoryRepository.findByCode(categoryCode);
+        return new CategoryFormDTO(categoryModel);
+    }
+
+    @Transactional
```

Comment:
Acho que essa anotação deveria ser no seu controller, assim você iria garantir que todo o fluxo vai estar dentro de uma transação.

Reply:
Dei uma pesquisada aqui e dizem que é melhor deixar na camada de serviço mesmo.

https://stackoverflow.com/questions/23118789/why-we-shouldnt-make-a-spring-mvc-controller-transactional
https://jike.in/qa/?qa=781553/